cmd/agent/agent.go

```diff
@@ -111,9 +110,15 @@ func run(c *cli.Context) error {
 	}
 	defer authConn.Close()
 
-	agentID := readAgentID(agentIDConfigPath)
+	agentConfig := readAgentConfig(agentConfigPath)
+
+	// deprecated
+	if agentConfig.AgentID == defaultAgentIDValue {
+		agentConfig.AgentID = readAgentID(c.String("agent-id-config-path"))
```

Comment:
If PR is the breaking changes, for why is this compatibility? Considering that these changes will be released in 1.0.0.

Reply:
Well people use next already so there is a migration path ... 

... so it is breaking, but not for the release ... thats why no mentioning in docs and changelog skip

We can remove depricated code soon too ...

Reply:
OK, then it's not a breaking change in current implementation. It's deprecation and some warning should be logged.
![image](https://github.com/woodpecker-ci/woodpecker/assets/127358482/8d757e85-a3ef-41b8-b406-8afc3832b309)

Reply:
well that wont appear either ... as it never got released in the first place ... so we dont have to document the removeall after this pull

Reply:
I'm just "nice" with the migration path, that will stick in for a few days before removal

Reply:
Can't  mark as resolved (don't know why). So, do it, if you can.